libkbfs/block_types.go

```diff
@@ -250,6 +250,9 @@ func (fb *FileBlock) Set(other Block) {
 	fb.Contents = fbCopy.Contents
 	fb.IPtrs = fbCopy.IPtrs
 	fb.ToCommonBlock().Set(fbCopy.ToCommonBlock())
+	if otherFb.hash != nil {
```

Comment:
Please put this in a comment here so we don't forget why we're doing it.  Though maybe expand it a little because I don't totally follow this comment.